storage/tests/system.py

```diff
@@ -952,3 +952,17 @@ def test_notification_w_user_project(self):
             self.assertEqual(notifications[0].topic_name, self.TOPIC_NAME)
         finally:
             notification.delete()
+
+
+class TestAnonymousClient(unittest.TestCase):
+
+    PUBLIC_BUCKET = 'gcp-public-data-landsat'
+
+    def test_access_to_public_bucket(self):
+        anonymous = storage.Client.create_anonymous_client()
+        bucket = anonymous.bucket(self.PUBLIC_BUCKET)
+        for blob in bucket.list_blobs():
+            print("Downloading blob: {}".format(blob.name))
+            with tempfile.TemporaryFile() as stream:
+                blob.download_to_file(stream)
+            break  # just use the first one
```

Comment:
Instead, just grab the first from the iterator:

```python
blobs_iter = bucket.list_blobs(max_results=1)
blob = next(blobs_iter)
```

Reply:
That no work:

```python
    blob = next(bucket.list_blobs(max_results=1))
TypeError: 'HTTPIterator' object is not an iterator
```

ea8244b does:

```python
blob, = list(bucket.list_blobs(max_results=1))
```